### ISSUE TITLE
如何在其他开发语言中使用Xlearn, 比较java

### ISSUE THREAD

**Issue by payne4handsome** [aksnzhy/xlearn#188]
模型训练完了以后我用setTXTModel保存了模型的参数，那么我在其他开发语言中，用fm的计算公式是可以得到最后的输出的，但是训练模型时候xlearn把训练数据归一化了，这样我在其他开发语言中就没有使用模型的参数得到模型的输出了。因为不知道归一化的方式。请问这种问题有没有解决方案？还是只能在顺利模型的不使用归一化？

**Comment by aksnzhy** [comment]
@payne4handsome xLearn can save the TXT model to disk file, and users can use any programming language to load the TXT data and use it in any way.

**Comment by payne4handsome** [comment]
@aksnzhy  No, you not understand me. If I train a model with setting option instance-wise normalization(default), so online application must normalize the input. Though I can use model parameters by TXT model in java, but I don't know how to normalize the input in java . Do you know what I mean? I have tested it. I can get right result when I train model with option --no-norm in python and then use model parameters in java by Txt model. On the contrary, get wrong result .

**Comment by aksnzhy** [comment]
@payne4handsome I see. This is a problem. For now, I think the only way to solve this problem is to implement instance-wise normal in Java. 

For instance-wise normal, given the following example:

`0:0.1 1:0.2 3:0.3 4:0.5`

The norm is equal to:

`norm = sqrt(0.1*0.1 + 0.2*0.2 + 0.3*0.3 + 0.4*0.4)`

Then we normalize the input data:

`0:0.1/norm 1:0.2/norm 2:0.3/norm 4:0.5/norm`

**Comment by payne4handsome** [comment]
@aksnzhy oh, thanks. I did as you said. I got a right result in java. I think I got a mistake. I think instance-wise normal mentioned in [official document](https://xlearn-doc.readthedocs.io/en/latest/python_api/index.html#instance-wise-normalization) is just like Min-max normalization or 0-score standardization.
+ Min-max normalization
![image](https://user-images.githubusercontent.com/28085731/49850922-614f6c80-fe1a-11e8-9448-6668c9358757.png)
+ 0-score standardization
![image](https://user-images.githubusercontent.com/28085731/49850982-93f96500-fe1a-11e8-9410-3619ee52be3b.png)

So, I have two other questions.
1.  If I have the following example:
0:0.1 1:1 3:100 4:1000 5:1000000
Before training  model used XLearn, should I normalize the input data like Min-max normalization or 0-score standardization,  If Xlearn not help us do , I think we should.
2. why use instance-wise normal that you mentioned by answered my last question. you say "**this technique is very useful**" in your document. Can you offer some reference material about instance-wise normal to prove it is very useful？
 
Thanks for your replying.

**Comment by delongwu** [comment]
norm = sqrt(0.1*0.1 + 0.2*0.2 + 0.3*0.3 + 0.4*0.4)

when norm is zero， the model does not work correctly， it print out nan
